libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m

```diff
@@ -112,16 +104,24 @@ + (void) removeSharedInstanceWithUser:(SFUserAccount *) userAccount {
 }
 
 - (void) dealloc {
-    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationDidEnterBackgroundNotification object:nil];
-    SFSDK_USE_DEPRECATED_BEGIN
-    [[SFAuthenticationManager sharedManager] removeDelegate:self];
-    SFSDK_USE_DEPRECATED_END
+    // Only work with auth-based notifications for an authenticated context.
+    if (_userAccount != nil) {
+        SFSDK_USE_DEPRECATED_BEGIN
+        [[SFAuthenticationManager sharedManager] removeDelegate:self];
+        SFSDK_USE_DEPRECATED_END
+    }
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (instancetype)init {
+    return [self initWithUser:nil];
 }
 
 - (instancetype) initWithUser:(SFUserAccount *) userAccount {
     self = [super init];
     if (self) {
-        self.userAccount = userAccount;
+        _userAccount = userAccount;
         SFSDKDeviceAppAttributes *deviceAttributes = [[self class] getDeviceAppAttributes];
         NSString *rootStoreDir = [[SFDirectoryManager sharedManager] directoryForUser:userAccount type:NSDocumentDirectory components:@[ kEventStoresDirectory ]];
```

Comment:
Shouldn't this be something along these lines:
```
NSString *rootStoreDir = nil;
if (_userAccount != nil) {
    NSString *rootStoreDir = [[SFDirectoryManager sharedManager] directoryForUser:userAccount type:NSDocumentDirectory components:@[ kEventStoresDirectory ]];
} else {
    NSString *rootStoreDir = [[SFDirectoryManager sharedManager] globalDirectoryOfType:NSDocumentDirectory components:@[ kEventStoresDirectory ]];
}
```

Reply:
I looked at the code in `SFDirectoryManager`, and it defaults to a global directory, without a user.  But that said, I agree that it's better to be more explicit.  I'll change it.